sklearn/covariance/_shrunk_covariance.py

```diff
@@ -484,12 +489,22 @@ def fit(self, X, y=None):
         # Not calling the parent object to fit, to avoid computing the
         # covariance matrix (and potentially the precision)
         X = self._validate_data(X)
+        if X.ndim == 1:
+            X = np.reshape(X, (1, -1))
+            warnings.warn(
+                "Only one sample available. You may want to reshape your data array"
+            )
+        if X.shape[0] == 1:
+            X = X.reshape(1, -1)
+            warnings.warn(
+                "Only one sample available. You may want to reshape your data array"
+            )
         if self.assume_centered:
             self.location_ = np.zeros(X.shape[1])
         else:
             self.location_ = X.mean(0)
         with config_context(assume_finite=True):
```

Comment:
Actually, we don't need the context manager anymore because we are not going to have a `check_array` anymore.

Reply:
I still think that we can remove this line since we don't have any `check_array` in the private function anymore